### ISSUE TITLE
KeepassXC-Browser: key exchange failed

### ISSUE THREAD

**Issue by JPT77** [keepassxreboot/keepassxc-browser#949]
I got a fresh installation of Kubuntu 20.04 and now I'm trying to connect keepassXC with Firefox. 
But it just doesn't work.

The file ~/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json was missing. I added it with the following content. I wasn't able to find the source file in the git repo, so I fetched it from the pinned issue. (so please add the official content of this file to the troubleshooting guide)

	{
		"allowed_extensions": [
			"keepassxc-browser@keepassxc.org"
		],
		"description": "KeePassXC integration with native messaging support",
		"name": "org.keepassxc.keepassxc_browser",
		"path": "/usr/bin/keepassxc-proxy",
		"type": "stdio"
	}

After adding this file the first error message changed: it now says "message encryption failed". when I click reload, it changes to the old message "key exchange failed" (both translated back to english from german)
Debugging console does not show anything else than error 7 and 9 plus ever repeating 

    Connecting to native messaging host org.keepassxc.keepassxc_browser keepass.js:984:13
    Error: No content script available for this tab. keepass.js:1188:25

the proxy is running, since I added the json it even got parameters:
/usr/bin/keepassxc-proxy /home/user/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json keepassxc-browser@keepassxc.org

KeePassXC - 2.4.3
KeePassXC-Browser - 1.6.6
Operating system: Kubuntu 20.04 64 bit, kernel 5.4.0-40-generic
Browser: Firefox 78.0.2

This database file was already connected to Firefox in my previous Kubuntu-installation.

any idea?

**Comment by varjolintu** [comment]
Your Firefox is probably installed via Snap. Use the official PPA or AppImage instead.

**Comment by JPT77** [comment]
no, it's from the ubtuntu repo

    firefox/focal-updates,focal-security,now 78.0.2+build2-0ubuntu0.20.04.1 amd64

**Comment by varjolintu** [comment]
Still, 2.4.3 is quite old compared to the latest 2.6.0 so I suggest updating it.

**Comment by JPT77** [comment]
installing 2.6.0-1ppa2~focal1 did not help either. :(
any way to make the debug log more detailed?

**Comment by varjolintu** [comment]
Please check the [Troubleshooting guide](https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide) and report your findings.

**Comment by JPT77** [comment]
Well, I already did. 
1. json file was missing, I added the one I found in the pinned issue. Would be nice to have a sample json in the troublshooting guide.
2. Ah, I missed the part about the json content.
Well, the text under 2) is hard to read and understand. Please improve. 
But, well, seems to be everything OK. 
Did you check if my json file is correct?
3. as I already said, proxy is running
4. debugging did not give any new insights.

**Comment by varjolintu** [comment]
>     1. json file was missing, I added the one I found in the pinned issue. Would be nice to have a sample json in the troublshooting guide.

If the JSON file is missing, did you enable Browser Integration from KeePassXC and enabled support for Firefox? That should do the installation automatically. No need for manual install.

Still, you can find an example file here:
https://github.com/keepassxreboot/keepassxc-browser/wiki/Native-messaging-script-files

> 
>     2. Ah, I missed the part about the json content.
>        Well, the text under 2) is hard to read and understand. Please improve.
>        But, well, seems to be everything OK.
>        Did you check if my json file is correct?
> 
>     3. as I already said, proxy is running

Your script is fine. Have you tried to delete existing keys from KeePassXC's Database settings? Do the same for the extension using the Connected Databases settings tab.

**Comment by JPT77** [comment]
No, I did not enable browser integration. That solved it. 

I think I found a link from FAQ? that pointed to a guide, but this link was dead and I wasn't able to find a guide.

is it ok if I add my findings to the troubleshooting guide? do you edit the guide through git?

**Comment by varjolintu** [comment]
Everything is already explained here: https://keepassxc.org/docs/

There's even a separate page for how to setup Browser Integration:
https://keepassxc.org/docs/KeePassXC_GettingStarted.html#_setup_browser_integration

So no need to add these basic instructions to the troubleshooting guide.

**Comment by JPT77** [comment]
Ok. But at least add a link to the top of the troubleshooting guide. then I wouldn't have had so many problems.

Thanks for your help.

**Comment by JPT77** [comment]
I tried to add the following to the troubleshooting guide, but it just doesn't work. So please could someone else?

	Before starting with trouble-shooting, did you follow the instrutions found here: 
	https://keepassxc.org/docs/#faq-browser
	https://keepassxc.org/docs/KeePassXC_GettingStarted.html#_setup_browser_integration

**Comment by varjolintu** [comment]
> I tried to add the following to the troubleshooting guide, but it just doesn't work. So please could someone else?
> 
> ```
> Before starting with trouble-shooting, did you follow the instrutions found here: 
> https://keepassxc.org/docs/#faq-browser
> https://keepassxc.org/docs/KeePassXC_GettingStarted.html#_setup_browser_integration
> ```

I'll add the links.
